### PR TITLE
Kp/04 rework

### PR DESCRIPTION
**Pull request by kp483**
Update PSE 04 to align with C17 format

### CODE REVIEW

File: 04/implement-palindrome.instructor.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        # Arrange
+        s = 'Kayak'
+
+        # Act
+        result = palindrome(s)
+
+        # Assert
+        assert result == True
+
+    def test_ignores_non_alphanumeric_chars(self):
```
Comment (beccaelenzil) [inline 04/implement-palindrome.instructor.md:53]:
```suggestion
    def test_raise_type_error_when_non_string_input():
        # Arrange
        s = []

        # Assert
        with pytest.raises(TypeError):
            palindrome(s) # Act

    def test_ignores_case():
        # Arrange
        s = 'Kayak'

        # Act
        result = palindrome(s)

        # Assert
        assert result == True

    def test_ignores_non_alphanumeric_chars():
```

**Review by beccaelenzil — APPROVED** [review]
LGTM! I like the edge case solution and tests :)
